delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/shacl/ShaclEngine.scala

```diff
@@ -0,0 +1,108 @@
+package ch.epfl.bluebrain.nexus.delta.rdf.shacl
+
+import java.net.URI
+import java.util
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxsh
+import monix.bio.IO
+import org.apache.jena.query.Dataset
+import org.apache.jena.rdf.model._
+import org.topbraid.jenax.util.{ARQFactory, JenaDatatypes}
+import org.topbraid.shacl.arq.SHACLFunctions
+import org.topbraid.shacl.engine.{Constraint, ShapesGraph}
+import org.topbraid.shacl.util.{SHACLSystemModel, SHACLUtil}
+import org.topbraid.shacl.validation.{ValidationEngine, ValidationEngineConfiguration, ValidationUtil}
+
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Extend the [[ValidationEngine]] form TopQuadrant in order to add triples to the report
+  * with the number of targetedNodes
+  *
+  * @param dataset        the Dataset to operate on
+  * @param shapesGraphURI the URI of the shapes graph (must be in the dataset)
+  * @param shapesGraph    the ShapesGraph with the shapes to validate against
+  */
+// $COVERAGE-OFF$
+@SuppressWarnings(Array("NullParameter"))
+final class ShaclEngine private (dataset: Dataset, shapesGraphURI: URI, shapesGraph: ShapesGraph)
```

Comment:
It would be nice to have tests on it also to see how it is being used

Reply:
This code does not need to be tested because it is a third party code which is tested by topQuadrant itself.
You'll have examples of this on `Resources` or `Schemas` afterwards

Reply:
I think it's worth adding a test to make sure that the local code will continue to function when doing lib updates